### ISSUE TITLE
Game Jolt Client unable to uninstall games

### ISSUE THREAD

**Issue by mg8973** [gamejolt/next-issue-tracker#1310]
**How are you using Game Jolt?**
Game Jolt Client (latest version as of today), Windows 10

**Please describe the issue you are having**
Whenever I try to uninstall certain games in the client, they usually fail for some reason, even when none of the game's processes are running.

The screenshot provided shows the error that the client displays when trying to uninstall a game:
![Screenshot (58)](https://user-images.githubusercontent.com/52803804/61090374-3dc1c400-a3f3-11e9-958b-04d4e87f42cc.png)

**Comment by promuch** [comment]
yeah same things happening with me on a different game.

**Comment by Sans2345** [comment]
I have the same thing

**Comment by YLivay** [comment]
Right after trying to uninstall, please scroll down to the footer and hit "Send System Report", maybe we'll see more info there.

**Comment by mg8973** [comment]
I've sent the system report.

**Comment by mg8973** [comment]
I also discovered another issue, when I try to delete the remaining game file (executable), it gives me this error message. (This happens with all games that were installed with the client)
![Capture](https://user-images.githubusercontent.com/52803804/62001173-221b1680-b09f-11e9-8fd5-5a29158f0fc8.PNG)

**Comment by YLivay** [comment]
Thats interesting. That means the game's executable must be open somehow. The client opens the game like every other normal executable, nothing fancy there so if you don't see the game itself it might be an issue with the game itself. Can you try closing the client completely and then trying to delete the file? Just to be sure it's the client that's holding the file and not some other program like an antivirus scanning it or something.

Usually, most of these issues end up being hyper active anti viruses interfering with how the client works. Anyways, i'm working on a new client release with better logging, when it's out i'll update the issue and ask you to submit another system report, ok?

At the moment the log you sent doesn't have anything useful to troubleshooting this issue sadly :(

**Comment by dancingdusekkar** [comment]
> I also discovered another issue, when I try to delete the remaining game file (executable), it gives me this error message. (This happens with all games that were installed with the client)
> ![Capture](https://user-images.githubusercontent.com/52803804/62001173-221b1680-b09f-11e9-8fd5-5a29158f0fc8.PNG)

This most likely means you have to go to task manager and end the process manually. It sucks, I know.

**Comment by YLivay** [comment]
@dancingdusekkar The client only opens the file for writing when downloading and updating the game. Theres nothing special in how the client executes the game either.

@pcgamer31098 So, which anti virus do you have?

**Comment by mg8973** [comment]
I use the free edition of BitDefender

**Comment by mg8973** [comment]
I installed a couple of games (including the one used in the screenshot) without using the client and I was able to successfully delete all of the game files after running them. Then I tried to uninstall the games using the Game Jolt Client again except I got a different result this time. I was able to delete the executable file, but the folder still remained. It also gave me an error similar to the one seen in my previous screenshot when I tried to delete it.

**Comment by mg8973** [comment]
I manually quit the Game Jolt Client and temporarily disabled my antivirus, but I still got the same result.

**Comment by YLivay** [comment]
The folder should remain. The client doesn't remove the folders in case there are files in there that were generated by the game (like saves). Something's fishy is going on in your setup if even after quitting the client the files are still in use. I doubt at this point it's client related. Either way, i'm working on pushing out a new version with better logging so i could see what's going on a bit better hopefully.

We'd be able to see which process is holding the file in the meantime:
Hit `Windows key + R`, it'll open a little run window. in it type `resmon` and hit enter.
It should've opened a program called `Resource Monitor`. Go to the `CPU` tab and in the `Associated Handles` section can you search for the name of the file/folder you're having issues deleting?
The `Image` field should say the name of the process holding your file hostage.

**Comment by mg8973** [comment]
I can't find any process that's interfering with this, the weird thing is that even when the game folder is completely empty, the game jolt client still says that it failed to install, the only way I got rid of that error was by manually deleting the game folder. (For some reason I didn't get the error earlier).

**Comment by YLivay** [comment]
That one is a different known issue. The client is expecting the manifest file to exist to know HOW to uninstall, but it doesnt fail gracefully when the directory is completely empty haha. The question is why the client reported it failed to uninstall the package when it actually succeeded. I do have a fix in the pipe for this one.

**Comment by mg8973** [comment]
Was the new version with better logging released, or if not, do you have a release date for it?

**Comment by YLivay** [comment]
@pcgamer31098 yes, it has. the latest client version is 0.34.0. If you are on a newer version you should be good to try again :)

**Comment by mg8973** [comment]
One thing that I forgot to mention is that even when I restart my PC, the same error still shows up in the client.

**Comment by YLivay** [comment]
I got your client log, gonna take a quick look now! thanks for the cooperation btw, its really helpful when tracking these elusive buggers!

**Comment by YLivay** [comment]
Woah it looks like communications between the client and the process that manages games is not working for you! This is most likely a firewall or a hyperactive anti virus of sorts.

Looks like this issue should be manifesting for any game related operation. Do you only have issues uninstalling packages? What about installing and launching games, does that work?

**Comment by YLivay** [comment]
The process responsible for managing games is executed by the client. To make sure the client always knows what's happening, the manager process waits for the client to establish communications with it before it does anything. If the client doesn't do that within a reasonable time (currently 2 seconds) it times out and aborts the operation, causing the errors you've been getting.

There are 3 possible reasons for this:
1. For some reason, the connection is blocked completely.
2. The initial connection is not blocked but is taking an unusual amount of time to connect.
3. The manager process crashes immediately after a connection is established AND we're not getting any logs about it for some reason.

For cases 1 and 2 it's almost always firewall or hyperactive anti virus as I mentioned before and im not sure theres much I can do about it besides trying to increase the timeout or rewriting how the client and the process communicates with each other.

For case 3 it's hard to tell why its crashing if its not outputting anything to our logs. It'd be really helpful if you could run the following command in a command prompt. Just press the windows key + R and in the run box that opens up type `cmd` and hit enter. A black window should appear. In it paste the following line:
```
%LOCALAPPDATA%\GameJoltClient\data-376715-735388\node_modules\client-voodoo\bin\GameJoltRunner.exe "--port" "63877" "--dir" "%LOCALAPPDATA%\game-jolt-client\User Data\Default\Games\sonic-time-twisted-24775\SonicTimeTwisted-133465" "--no-self-update" "uninstall"
```
You can paste what it output for you and send it to yariv@gamejolt.com and maybe it'll shed some more light on this pesky issue!

**Comment by mg8973** [comment]
I don't have any issues installing or launching games, I only have issues uninstalling them.

**Comment by mg8973** [comment]
Also the command prompt command doesn't output anything for me

**Comment by YLivay** [comment]
hmm.. maybe on prod builds the output is silenced without the debug flag. try this:
```
%LOCALAPPDATA%\GameJoltClient\data-376715-735388\node_modules\client-voodoo\bin\GameJoltRunner.exe "--port" "63877" "--dir" "%LOCALAPPDATA%\game-jolt-client\User Data\Default\Games\sonic-time-twisted-24775\SonicTimeTwisted-133465" "--no-self-update" "--debug" "uninstall"
```

**Comment by mg8973** [comment]
I sent the command prompt log via email

**Comment by mg8973** [comment]
Did the log help in any way?

**Comment by mg8973** [comment]
Any updates on this?

**Comment by mg8973** [comment]
I think I might've found out the cause of this issue, I tried installing and uninstalling some games with and without a volume mixer app called EarTrumpet running, and the games were successfully uninstalled when the app wasn't running. This is a known issue for EarTrumpet https://github.com/File-New-Project/EarTrumpet/issues/273 . Though I don't know why @promuch and @Sans2345 are having this issue as well.

**Comment by Matthew0006** [comment]
The same exact thing is happening to me on three different games
